new-lamassu-admin/src/pages/Commissions/helper.js

```diff
@@ -345,12 +347,181 @@ const getOrder = ({ machine, cryptoCurrencies }) => {
   return 3
 }
 
+const createCommissions = (cryptoCode, deviceId, isDefault, config) => {
+  return {
+    minimumTx: config.minimumTx,
+    fixedFee: config.fixedFee,
+    cashOut: config.cashOut,
+    cashIn: config.cashIn,
+    machine: deviceId,
+    cryptoCurrencies: [cryptoCode],
+    default: isDefault,
+    id: v4()
+  }
+}
+
+const getCommissions = (cryptoCode, deviceId, config) => {
+  const overrides = R.prop('overrides', config)
+
+  if (overrides && !R.isEmpty(overrides)) {
+    const specificOverride = R.find(
+      it =>
+        it.machine === deviceId && _.includes(cryptoCode)(it.cryptoCurrencies)
+    )(overrides)
+
+    if (specificOverride !== undefined)
+      return createCommissions(cryptoCode, deviceId, false, specificOverride)
+
+    const machineOverride = R.find(
+      it =>
+        it.machine === deviceId && _.includes('ALL_COINS')(it.cryptoCurrencies)
+    )(overrides)
+
+    if (machineOverride !== undefined)
+      return createCommissions(cryptoCode, deviceId, false, machineOverride)
+
+    const coinOverride = R.find(
+      it =>
+        it.machine === 'ALL_MACHINES' &&
+        _.includes(cryptoCode)(it.cryptoCurrencies)
+    )(overrides)
+
+    if (coinOverride !== undefined)
+      return createCommissions(cryptoCode, deviceId, false, coinOverride)
+  }
+
+  return createCommissions(cryptoCode, deviceId, true, config)
+}
+
+const getListCommissionsSchema = () => {
+  return Yup.object().shape({
+    machine: Yup.string()
+      .label('Machine')
+      .required(),
+    cryptoCurrencies: Yup.array()
+      .label('Crypto Currency')
+      .required(),
+    cashIn: Yup.number()
+      .label('Cash-in')
+      .min(0)
+      .max(percentMax)
+      .required(),
+    cashOut: Yup.number()
+      .label('Cash-out')
+      .min(0)
+      .max(percentMax)
+      .required(),
+    fixedFee: Yup.number()
+      .label('Fixed Fee')
+      .min(0)
+      .max(currencyMax)
+      .required(),
+    minimumTx: Yup.number()
+      .label('Minimum Tx')
+      .min(0)
+      .max(currencyMax)
+      .required()
+  })
+}
+
+const getTextStyle = (obj, isEditing) => {
+  return { color: obj.default ? '#1b2559' : '#44e188' }
```

Comment:
There are probably named variables for these colors, import them instead of hard coding like this

Reply:
Done